marlowe-playground-client/src/Wallet.purs

```diff
@@ -0,0 +1,1301 @@
+renderContract state contract =
+  li
+    [ ( classes
+          ( [ ClassName "contract-link" ]
+              <> if (state ^? (_walletLoadedContract <<< _Just <<< to _.idx)) == Just contract.idx then
+                  [ active ]
+                else
+                  [ ClassName mempty ]
+          )
+      )
+    ]
+    [ a [ onClick $ const $ Just $ SelectContract contract ] [ text $ "Contract " <> (show contract.idx) ]
+    ]
+
+inputItem ::
+  forall p.
+  Boolean ->
+  PubKey ->
+  ActionInput ->
+  HTML p Action
+inputItem isEnabled person (DepositInput accountId party token value) =
+  li [ classes [ aHorizontal ] ]
+    [ p_ (renderDeposit accountId party token value)
+    , button
+        [ classes [ plusBtn, smallBtn, (Classes.disabled $ not isEnabled) ]
+        , enabled isEnabled
+        , onClick $ const $ Just
+            $ AddInput (Just person) (IDeposit accountId party token value) []
+        ]
+        [ text "+" ]
+    ]
+
+inputItem isEnabled person (ChoiceInput choiceId@(ChoiceId choiceName choiceOwner) bounds chosenNum) =
+  li
+    [ classes [ aHorizontal, ClassName "flex-wrap", ClassName "choice-row" ] ]
+    [ div []
+        [ p [ class_ (ClassName "choice-input") ]
+            [ spanText "Choice "
+            , b_ [ spanText (show choiceName) ]
+            , spanText ": Choose value "
+            , marloweActionInput isEnabled (SetChoice choiceId) chosenNum
+            ]
+        , p [ class_ (ClassName "choice-error") ] error
+        ]
+    , button
+        [ classes [ plusBtn, smallBtn, if (isEnabled && inBounds chosenNum bounds) then (ClassName mempty) else Classes.hide ]
+        , enabled (isEnabled && inBounds chosenNum bounds)
+        , onClick $ const $ Just
+            $ AddInput (Just person) (IChoice (ChoiceId choiceName choiceOwner) chosenNum) bounds
+        ]
+        [ text "+" ]
+    ]
+  where
+  error = if inBounds chosenNum bounds then [] else [ text boundsError ]
+
+  boundsError = "Choice must be between " <> intercalate " or " (map boundError bounds)
+
+  boundError (Bound from to) = show from <> " and " <> show to
+
+inputItem isEnabled person NotifyInput =
+  li
+    [ classes [ ClassName "choice-a", aHorizontal ] ]
+    [ p_ [ text "Notify Contract" ]
+    , button
+        [ classes [ plusBtn, smallBtn, (Classes.disabled $ not isEnabled) ]
+        , enabled isEnabled
+        , onClick $ const $ Just
+            $ AddInput (Just person) INotify []
+        ]
+        [ text "+" ]
+    ]
+
+marloweActionInput :: forall p a action. Show a => Boolean -> (BigInteger -> action) -> a -> HTML p action
+marloweActionInput isEnabled f current =
+  input
+    [ type_ InputNumber
+    , enabled isEnabled
+    , placeholder "BigInteger"
+    , class_ $ ClassName "action-input"
+    , value $ show current
+    , onValueChange
+        $ ( \x ->
+              Just
+                $ f
+                    ( fromMaybe zero (BigInteger.fromString x)
```

Comment:
Same `Just...fromMaybe` suggestion as before.

Reply:
done.

Reply:
except this time I think `(map f <<< BigInteger.fromString)` looks better